WordPress/src/main/java/org/wordpress/android/datasets/ReaderDatabase.java

```diff
@@ -131,10 +133,27 @@ public void onOpen(SQLiteDatabase db) {
     /*
      * resets (clears) the reader database
      */
-    public static void reset() {
+    public static void reset(boolean retainBookmarkedPosts) {
         // note that we must call getWritableDb() before getDatabase() in case the database
         // object hasn't been created yet
         SQLiteDatabase db = getWritableDb();
+
+        if (retainBookmarkedPosts && ReaderPostTable.hasBookmarkedPosts()) {
+            ReaderTagList tags = ReaderTagTable.getBookmarkTags();
+            if (!tags.isEmpty()) {
+                ReaderPostList bookmarkedPosts = ReaderPostTable.getPostsWithTag(tags.get(0), 0, true);
```

Comment:
We might want to set the "excludeTextColumn" parameter to false as we need the text for offline support. wdyt?

Reply:
If it's a part of security measurements, feel free to merge this PR;).

Reply:
> We might want to set the "excludeTextColumn" parameter to false

Whoops, I had that reversed by mistake - glad you caught it. I fixed that and will merge when Travis is done.